docs/DeveloperGuide.md

```diff
@@ -1,37 +1,300 @@
 
+> **⚠️Warning⚠️**
+> - This tells you that there is some **important** information you should take note of to prevent issues from arising
+    when you are using the application.  
 
-## Product scope
-### Target user profile
+Click on the hyperlinks in the [content page](#Content-Page) to quickly navigate the developer's guide.
 
-{Describe the target user profile}
+## Design
+### Technologies used
+**MindMyMoney** is written fully in **Java 11** using  Object-Oriented Programming (OOP) paradigm to help structure and
+organise the code. This enables the efficiency of future improvements and revisions.
+Data stored in the application is saved into text files locally on the user's device.  
 
-### Value proposition
+### Architecture Overview
+![architecture diagram](images/architecture_diagram.png)  
+Fig 1 - Architecture Diagram for MindMyMoney
 
-{Describe the value proposition: what problem does it solve?}
+The Architecture Diagram above shows the high-level design of the application. The **main components**
+consist of:
+- `MMM`: Initialises the components in the correct sequence and connects them with each other. Also holds the user's
+expenditures in memory.
+- `Ui`: The User Interface of the application and deals with interaction with the user.
+- `Parser`: Deals with making sense of user commands.
+- `Commands`: The collection of all executable commands.
+- `Storage`: Reads data from, and writes data to the hard disk.
 
-## User Stories
+By abstracting out closely related code into classes, it allows `MMM` to deal at a higher level, without worrying
+about the lower level details. Higher cohesion is also achieved and coupling is minimized as each component is
+only coupled to the main class, `MMM`.
+
+The Sequence Diagram below shows an example of how the components interact with each other for the scenario 
+where the user issues the command`add shoes 100`.
+
+![sequence_diagram](images/sequence_diagram.png)
+<br/> Fig 2 - Sequence Diagram for MindMyMoney
+
+### Component Overview 
+The major code is broken down into components for better abstraction. 
+The sections below give more details for each component.
+
+### UI component
+The source code can be found in [`Ui.java`](https://github.com/AY2122S2-CS2113T-T10-4/tp/blob/master/src/main/java/seedu/mindmymoney/Ui.java)
+
+![ui_diagram](images/ui_diagram.png)
+<br/> Fig 3 - Ui Class Diagram
+
+The UI component consists of a `Ui` and `PrintStrings` class.
+
+The UI component:
+- Prints the welcome banner and message on startup, as well as a financial tip from the `PrintStrings` class.
+- Prints `>` before the user's input, to help the user differentiate between their input and an output from the
+application.
+- Reads input from the user.
+
+<br/>
+
+### Parser component
+The source code can be found in [`Parser.java`](https://github.com/AY2122S2-CS2113T-T10-4/tp/blob/master/src/main/java/seedu/mindmymoney/Parser.java)
+
+![parser_diagram](images/parser_diagram.png)
```

Comment:
Would it be possible to clarify this diagram? It is a bit confusing because a package named Parser is used. But the code architecture does not have a package named parser. Would it be possible to rename it to a more suitable name like ExpenditureTracker? Because it houses not only parser, but also the other aspects like the ExpenditureList, Functions as well as Parser